### PR TITLE
Migrate tools to null safety

### PR DESCRIPTION
**Pull request by amanv8060**
Migrates tool/ & gallery options to null safety.

part of #527

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: windows/flutter/generated_plugins.cmake [hunk 1]
```diff
@@ -6,6 +6,9 @@ list(APPEND FLUTTER_PLUGIN_LIST
   url_launcher_windows
 )
```
Comment (amanv8060) [inline windows/flutter/generated_plugins.cmake:7]: @guidezpl I don't understand why and how this file got generated, can you please guide me on this?

File: lib/codeviewer/code_segments.dart [hunk 2]
```diff
@@ -44312,37 +44312,30 @@ class CodeSegments {
...
+          style: codeStyle.stringStyle,
+          text: '\u0027package:flutter/material.dart\u0027'),
       TextSpan(style: codeStyle.punctuationStyle, text: ';'),
       TextSpan(style: codeStyle.baseStyle, text: '\u000a'),
       TextSpan(style: codeStyle.keywordStyle, text: 'import'),
       TextSpan(style: codeStyle.baseStyle, text: ' '),
-      TextSpan(style: codeStyle.stringStyle, text: '\u0027dart:ui\u0027'),
-      TextSpan(style: codeStyle.baseStyle, text: ' show '),
-      TextSpan(style: codeStyle.classStyle, text: 'Vertices'),
+      TextSpan(
```
Comment (amanv8060) [inline lib/codeviewer/code_segments.dart:44321]: These changes are being generated in a version before this change as well (after updating code-segments) : 

```dart
    TextSpan(
          style: codeStyle.stringStyle,
          text: '\u0027package:flutter/material.dart\u0027'),
      TextSpan(style: codeStyle.punctuationStyle, text: ';'),
      TextSpan(style: codeStyle.baseStyle, text: '\u000a'),
      TextSpan(style: codeStyle.keywordStyle, text: 'import'),
      TextSpan(style: codeStyle.baseStyle, text: ' '),
      TextSpan(
          style: codeStyle.stringStyle,
          text:
              '\u0027package:flutter_gen/gen_l10n/gallery_localizations.dart\u0027'),
      TextSpan(style: codeStyle.punctuationStyle, text: ';'),
      TextSpan(style: codeStyle.baseStyle, text: '\u000a\u000a'),
```

which should not be there IMO. 

Imports in hosted version : 
before

Imports after this change :
after

File: windows/flutter/generated_plugins.cmake (hunk 1 above)
Comment (amanv8060) [reply windows/flutter/generated_plugins.cmake:7]: pub get with latest flutter master , automatically adding it

File: lib/codeviewer/code_segments.dart (hunk 2 above)
Comment (guidezpl) [reply lib/codeviewer/code_segments.dart:44321]: I can't find the issue, but for some unknown reason, the code viewer generates differently on non-Unix platforms. Can you revert the changes to the the code-viewer?

File: windows/flutter/generated_plugins.cmake (hunk 1 above)
Comment (guidezpl) [reply windows/flutter/generated_plugins.cmake:7]: That's fine

File: lib/codeviewer/code_segments.dart (hunk 2 above)
Comment (amanv8060) [reply lib/codeviewer/code_segments.dart:44321]: yeah, will revert the code-segments changes.

Comment (amanv8060) [reply lib/codeviewer/code_segments.dart]: @guidezpl can you review and add these.

**Pull request merged**